base/shell/explorer/trayprop.cpp

```diff
@@ -196,11 +196,22 @@ class CStartMenuSettingsPage : public CPropertyPageImpl<CStartMenuSettingsPage>
         HWND hwndCustomizeClassic = GetDlgItem(IDC_TASKBARPROP_STARTMENUCLASSICCUST);
         HWND hwndCustomizeModern = GetDlgItem(IDC_TASKBARPROP_STARTMENUCUST);
         HWND hwndStartBitmap = GetDlgItem(IDC_TASKBARPROP_STARTMENU_BITMAP);
+        HWND hwndModernRadioBtn = GetDlgItem(IDC_TASKBARPROP_STARTMENU);
+        BOOL policyNoSimpleStartMenu = (int)SHRestricted(REST_NOSTARTPANEL) == 1 ? TRUE : FALSE;
```

Comment:
It always returns 0

Reply:
Don't check for `== 1`,
use either:
`BOOL policyNoSimpleStartMenu = SHRestricted(REST_NOSTARTPANEL) ? TRUE : FALSE;`
or:
`BOOL policyNoSimpleStartMenu = SHRestricted(REST_NOSTARTPANEL) != 0 ? TRUE : FALSE;`

Reply:
Updated

Reply:
Actually `? TRUE : FALSE` is an anti-pattern, it should be avoided. Just `SHRestricted(...) != 0` is a boolean expression -- no conditional operator needed.

Reply:
Updated